packages/health-check/src/checks/verbose.js

```diff
@@ -551,53 +551,47 @@ function uniswapIndexFileCheck(done) {
   skylinkVerification(done, linkInfo);
 }
 
-// skylinkVerification verifies a skylink against known information provided in
-// the linkInfo.
-function skylinkVerification(done, linkInfo) {
+// verifies a skylink against provided information
+function skylinkVerification(done, { name, skylink, bodyHash, metadata }) {
   const time = process.hrtime();
 
   // Create the query for the skylink
-  const query = `http://${process.env.PORTAL_URL}/${linkInfo.skylink}?nocache=true`;
+  const query = `http://${process.env.PORTAL_URL}/${skylink}?nocache=true`;
 
   // Get the Skylink
   superagent
     .get(query)
     .responseType("blob")
-    .end((err, res) => {
-      // Record the statusCode
-      const statusCode = (res && res.statusCode) || (err && err.statusCode) || null;
-      let info = null;
-
-      // Determine if the skylink is up. Start with checking if there was an
-      // error in the request.
-      let up = err === null;
-      if (up) {
-        // Check if the response body is valid by checking against the known
-        // hash
-        const validBody = hash(res.body) === linkInfo.bodyHash;
-        // Check if the metadata is valid
-        const metadata = res.header["skynet-file-metadata"] ? JSON.parse(res.header["skynet-file-metadata"]) : null;
-        const validMetadata = isEqual(metadata, linkInfo.metadata);
-        // Redetermine if the Skylink is up based on the results from the body
-        // and metadata hash checks
-        up = up && validBody && validMetadata;
-
-        info = {
-          body: { valid: validBody },
-          metadata: { valid: validMetadata, diff: detailedDiff(metadata, linkInfo.metadata) },
-        };
-      }
+    .then(
+      (response) => {
+        const entry = { name, up: true, statusCode: response.statusCode, time: getTimeDiff(time) };
+        const info = {};
+
+        // Check if the response body is valid by checking against the known hash
+        if (hash(response.body) !== bodyHash) {
+          entry.up = false;
+          info.body = { valid: false, hash: { expected: bodyHash, current: hash(response.body) } };
+        }
+
+        // Check if the metadata is valid by deep comparing expected value with response
+        const expectedMetadata =
+          response.header["skynet-file-metadata"] && JSON.parse(response.header["skynet-file-metadata"]);
+        if (!isEqual(expectedMetadata, metadata)) {
+          entry.up = false;
+          info.metadata = { valid: false, diff: detailedDiff(expectedMetadata, metadata) };
+        }
+
+        if (!isEmpty(info)) entry.info = info; // add info only if it exists
```

Comment:
lodash's `isEmpty` is evil 😛  no need to change it though but ideally we get rid of that stupid dependency